gcp/appengine/frontend_handlers.py

```diff
@@ -651,3 +652,35 @@ def list_packages(vuln_affected: list[dict]):
 def not_found_error(error: exceptions.HTTPException):
   logging.info('Handled %s - Path attempted: %s', error, request.path)
   return render_template('404.html'), 404
+
+
+@blueprint.app_template_filter('has_link_to_deps_dev')
+def has_link_to_deps_dev(ecosystem):
+  """
+  Check if a given ecosystem has a corresponding link in deps.dev.
+
+  Returns:
+      bool: True if the ecosystem has a corresponding link in deps.dev,
+            False otherwise.
+  """
+  return osv.ecosystems.is_supported_in_deps_dev(ecosystem)
+
+
+@blueprint.app_template_filter('link_to_deps_dev')
+def link_to_deps_dev(package, ecosystem):
+  """
+  Generate a link to the deps.dev page for a given package in the specified
+  ecosystem.
+
+  Args:
+      package (str): The name of the package.
+      ecosystem (str): The ecosystem name.
+  Returns:
+      str or None: The URL to the deps.dev page for the package if the
+      ecosystem is supported, None otherwise.
+  """
+  system = osv.ecosystems.map_ecosystem_to_deps_dev(ecosystem)
+  if not system:
+    return None
+  encoded_package = parse.quote(package, safe='')
```

Comment:
https://github.com/google/osv.dev/pull/2277/files#r1639220513 seems worthy of being a comment in the code...

```suggestion
    # This ensures that special characters such as / are properly encoded, 
    # preventing invalid paths and 404 errors. 
    # e.g. for the package name github.com/rancher/wrangler, 
    # return https://deps.dev/go/github.com%2Francher%2Fwrangler
    encoded_package = parse.quote(package, safe='')
```